### DISCUSSION TITLE
Whats the best way to display the battery of your accessories?

### DISCUSSION THREAD

**Discussion by Karlzzon** [i3/i3#5918] [Q&A]
Hi,
I was wondering if there is any good way to display the battery life of my accessories?
Ive got a hyperX cloud alpha and a logitech superlight, and im currently using i3block as my topbar, but im open for suggestions on switching that up.

**Comment by orestisfl** [comment]
Do they show up under `/sys/class/power_supply` or `acpi -b`? With i3blocks you can have multiple battery blocks, each showing a different battery level.

**Reply by Karlzzon** [reply]
power_supply is empty and acpi -b shows : No support for device typ: power_supply

**Reply by orestisfl** [reply]
Hard to know without owning the same hardware but some random things that could lead to a solution.

First, for logitech, there is the [Solaar](https://github.com/pwr-Solaar/Solaar) project.

There is also the option of booting into Windows, running G Hub and checking for firmware updates for your hardware.

To debug your current system, the output of these might be helpful:
```
uname -a
sudo dmesg | grep -i acpi
lsmod | grep -i acpi
```

**Reply by Karlzzon** [reply]
So the goal is making sure the devices show up in either power_supply or acpi -b  ? Just making sure im understanding you correctly.
Once i get them to, i should be able to add them to i3blocks without issues?

**Reply by orestisfl** [reply]
If they do show up in acpi it would be trivial to add them to i3bar, but I am not sure if it's the only (or even possible) approach. I realized that my own bluetooth headphones don't show up under `acpi -b` but I found a different way to get the percentage, source: https://stackoverflow.com/a/55008142

What I did first was:
```
upower --dump
```
scroll to my headphones, copy the `native-path` and then run:
```
dbus-send --print-reply=literal --system --dest=org.bluez '<native-path>' org.freedesktop.DBus.Properties.Get string:"org.bluez.Battery1" string:"Percentage"
```
which reports the battery level.

**Reply by Karlzzon** [reply]
Ah, the only thing i get from upower --dump is : 
```
upower --dump
Device: /org/freedesktop/UPower/devices/DisplayDevice
  power supply:         no
  updated:              Tue 13 Feb 2024 12:40:46 CET (118 seconds ago)
  has history:          no
  has statistics:       no
  unknown
    warning-level:       none
    percentage:          0%
    icon-name:          'battery-missing-symbolic'

Daemon:
  daemon-version:  0.99.20
  on-battery:      no
  lid-is-closed:   no
  lid-is-present:  no
  critical-action: PowerOff
```

**Reply by orestisfl** [reply]
Maybe from `bluetoothctl` then?

```
$ bluetoothctl devices
Device AA:BB:CC:DD:EE:FF My headphones
```

and then:
```
dbus-send --print-reply=literal --system --dest=org.bluez '/org/bluez/hci0/dev_AA_BB_CC_DD_EE_FF' org.freedesktop.DBus.Properties.Get string:"org.bluez.Battery1" string:"Percentage"
```

**Reply by Karlzzon** [reply]
I ran lsusb and got

```
Bus 001 Device 003: ID 03f0:098d HP, Inc HyperX Cloud Alpha Wireless
Bus 001 Device 002: ID 046d:c547 Logitech, Inc. USB Receiver
```
but got nothing from bluetoothctl. That ID isnt the right one is it?

**Reply by orestisfl** [reply]
Are your devices connected over bluetooth or just with the usb dongle? If the second option, then the bluetoothctl approach won't work.

**Reply by Karlzzon** [reply]
usb dongle for both. Sorry

**Reply by orestisfl** [reply]
No idea then :/ Try https://github.com/pwr-Solaar/Solaar for your logitech mouse

**Reply by Karlzzon** [reply]
Ait ait, Thank you for your time
